### PR TITLE
goal: Change asset commands to use AccountAssetInformation

### PR DESCRIPTION
**Pull request by algochoi**
## Summary

This PR addresses some open comments made in https://github.com/algorand/go-algorand/pull/4684

Addresses three things:
* Change `EncodedBlockCert` decoding to use `Decode` instead of `DecodeReflect`
* Use `AccountAssetInformation` API in `goal asset info`. This changes behavior in `goal asset info` that looked like a bug regarding reserve addresses: the command now returns an error if a reserve asset is not opted into the asset, whereas previously, the command returned zero-ed values without failing. One test was changed to reflect this behavior.
* Refactor `libgoal` to use `AccountAssetInformation` instead of `AccountInformationV2`

## Test Plan

CI tests.

### CODE REVIEW

File: libgoal/transactions.go [hunk 1]
```diff
@@ -717,24 +717,14 @@ func (c *Client) MakeUnsignedAssetConfigTx(creator string, index uint64, newMana
 		params = asset.Params
 	} else {
 		// Fetch the current state, to fill in as a template
-		current, err := c.AccountInformationV2(creator, true)
+		current, err := c.AccountAssetInformation(creator, index)
```
Comment (jasonpaulos) [inline libgoal/transactions.go:720]: The change here is already an improvement, but I think the entire if/else statement can be removed and replaced with the contents of the current if branch (`c.AssetInformationV2`). Perhaps plus an additional assertion that if `creator` is not empty, it matches the asset creator.

**Review by jasonpaulos — COMMENTED** [review]
Thanks for taking the time to make these improvements.

I left a suggestion for some additional code simplification, but consider it optional.

File: libgoal/transactions.go (hunk 1 above)
Comment (algochoi) [reply libgoal/transactions.go]: Good point - I took this suggestion in [8e97a9f](https://github.com/algorand/go-algorand/pull/4758/commits/8e97a9f927c54809cf8fe91ef98896ddf0fff7d5)

**Review by jasonpaulos — DISMISSED** [review]

File: test/scripts/e2e_subs/asset-misc.sh [hunk 2]
```diff
@@ -102,6 +102,16 @@ fi
 # case 3: asset created with manager, reserve, freezer, and clawback different from the creator
 ${gcmd} asset create --creator "${ACCOUNT}" --manager "${ACCOUNTB}" --reserve "${ACCOUNTC}" --freezer "${ACCOUNTD}" --clawback "${ACCOUNTE}" --name "${ASSET_NAME}" --unitname dma --total 1000000000000 --asseturl "${ASSET_URL}"
 
+# case 3a: asset info should fail if reserve address has not opted into the asset.
+if $(${gcmd} asset info --creator $ACCOUNT --unitname dma); then
+    date '+asset-misc FAIL asset info should fail unless reserve account was opted in %Y%m%d_%H%M%S'
+    exit 1
+else
+    echo ok
```
Comment (michaeldiamant) [inline test/scripts/e2e_subs/asset-misc.sh:110]: @algochoi How involved is it to assert the error is the expected error?

Sorry for the late remark.  If it's onerous, I'm willing to merge as is though it feels unsafe.

Comment (algochoi) [reply test/scripts/e2e_subs/asset-misc.sh:112]: I added an assert for the error message as well

Comment (michaeldiamant) [reply test/scripts/e2e_subs/asset-misc.sh:112]: @algochoi Resolving - Thanks for taking in the feedback.

**Review by michaeldiamant — APPROVED** [review]

**Comment by michaeldiamant** [comment]
Merging - I reviewed the backward incompatible API change and agree with categorizing it as a bug fix.

**Pull request merged**